scripts/apidocgen/generate.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+# This script generates swagger description file and required Go docs files
+# from the coderd API.
+
+set -euo pipefail
+
+SCRIPT_DIR=$(dirname "${BASH_SOURCE[0]}")
+API_MD_TMP_FILE=$(mktemp /tmp/coder-apidocgen.XXXXXX)
+
+cleanup() {
+	rm -f "${API_MD_TMP_FILE}"
+}
+trap cleanup EXIT
+
+echo "Use temporary file: ${API_MD_TMP_FILE}"
+
+(
```

Comment:
Why put these in subshells? If you're using subshells just for changing directories you should use `pushd` and `popd` instead like so:

```
pwd # /home/dean
pushd "./dir"
pwd # /home/dean/dir
popd
pwd # /home/dean
```

Reply:
Fixed